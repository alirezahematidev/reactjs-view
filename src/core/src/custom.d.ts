declare module "*.png" {
  const value: any;
  export default value;
}

declare module "*.svg" {
  const content: any;
  export default content;
}

declare namespace globalThis {
  export const __DEV__: boolean;
  export const __MOCK__: boolean;
  export interface CSSRule {
    style: any;
    styleMap: any;
    selectorText: string;
  }
}
