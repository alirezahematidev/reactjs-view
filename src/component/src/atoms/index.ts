export * from "./anchorView";
export * from "./view";
