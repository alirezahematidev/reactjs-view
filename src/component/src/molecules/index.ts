export * from "./anchorView";
export * from "./loadingView";
export * from "./scrollView";
export * from "./text";
