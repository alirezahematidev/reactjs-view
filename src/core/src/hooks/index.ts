export * from "./useDebounce";
export * from "./useElementLayout";
export * from "./useInteractions";
export * from "./useIsomorphicLayoutEffect";
