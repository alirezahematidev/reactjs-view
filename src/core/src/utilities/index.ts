export * from "./generateIndex";
export * from "./ref";
