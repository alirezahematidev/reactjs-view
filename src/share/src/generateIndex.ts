type Index =
  | "atoms"
  | "molecules"
  | "muscles"
  | "organisms"
  | "templates"
  | "pages";

const getIndexes = (startIndex: number) => ({
  atoms: startIndex + 1000,
  molecules: startIndex + 3000,
  muscles: startIndex + 5000,
  organisms: startIndex + 7000,
  templates: startIndex + 9000,
  pages: startIndex + 11000,
});

const scopes = {
  coreModule: getIndexes(20000),
  module: getIndexes(40000),
  app: getIndexes(60000),
};

type Scope = keyof typeof scopes;

export type { Scope, Index };
export { scopes };
