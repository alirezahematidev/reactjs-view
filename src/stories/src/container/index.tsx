import { View } from "reactjs-view";
import "./index.css";
import useStyles from "./style";

export const StoryContainer = ({ children }: any) => {
  const classes = useStyles();
  return <View className={classes.layoutContainer}>{children}</View>;
};
