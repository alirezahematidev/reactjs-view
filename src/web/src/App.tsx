import { Route, Routes } from "react-router-dom";
import { View } from "reactjs-view";
import { AppProviders } from "./appProviders";

const App = () => {
  return (
    <AppProviders>
      <Routes>
        <Route
          path="*"
          element={
            <View style={{ flex: 1 }}>
              <View style={{ flex: 1 }}>
                <Routes>
                  <Route path="/team" element={<div>team</div>} />
                  <Route path="/user" element={<div>user</div>} />
                  <Route path="/timeline" element={<div>timeline</div>} />
                  <Route path="/screen" element={<div>screen</div>} />
                  <Route
                    path="/productivity"
                    element={<div>productivity</div>}
                  />
                </Routes>
              </View>
            </View>
          }
        />
      </Routes>
    </AppProviders>
  );
};

export default App;
