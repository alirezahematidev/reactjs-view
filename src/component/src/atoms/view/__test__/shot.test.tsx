import { shotTest } from "reactjs-view-core/src/tests";
import { View } from "../";

shotTest(View, {
  href: "http://google.com",
  onPress: () => {},
  className: "className",
  style: {},
});
