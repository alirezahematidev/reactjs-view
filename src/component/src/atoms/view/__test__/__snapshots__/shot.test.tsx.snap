// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mount and unmount component could be updated and unmounted without errors 1`] = `
<DocumentFragment>
  <div
    class="container-0-2-1"
  />
</DocumentFragment>
`;

exports[`mount and unmount component could be updated and unmounted without errors 2`] = `
<DocumentFragment>
  <a
    class="container-0-2-1 pressable-0-2-2 className"
    href="http://google.com"
  />
</DocumentFragment>
`;
